pkg/providers/snow/reconciler/reconciler_test.go

```diff
@@ -0,0 +1,95 @@
+package reconciler_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/aws/eks-anywhere/controllers/resource/mocks"
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/providers/snow/reconciler"
+)
+
+func TestReconcileSuccess(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	cluster := &v1alpha1.Cluster{}
+	client := mocks.NewMockClient(ctrl)
```

Comment:
This is not blocking but we don't need to use mocks, we can use the kube fake client
I personally think it makes writing tests for controller code easier. But if you think otherwise, I'm good with this

Reply:
Good idea, it's simpler to use the built-in fake client.